### PR TITLE
Prefer TrinoException over RTE in MysqlClient

### PR DESCRIPTION
**Pull request by kokosing**
Prever TrinoException over RTE in MysqlClient

Fixes https://github.com/trinodb/trino/issues/7208

### CODE REVIEW

**Comment by wendigo** [comment]
s/Prever/Prefer/g

**Comment by wendigo** [comment]
Thx @kokosing

**Comment by kokosing** [comment]
@findepi @wendigo PTAL

**Comment by findepi** [comment]
https://github.com/trinodb/trino/pull/7212#issuecomment-792729837

**Comment by kokosing** [comment]
AC

**Review by wendigo — APPROVED** [review]

File: plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMysqlWithoutDataSource.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+public class TestMysqlWithoutDataSource
+        extends AbstractTestQueryFramework
+{
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put("connection-url", "jdbc:mysql://not_existing_mysql")
```
Comment (findepi) [inline plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMysqlWithoutDataSource.java:36]: underscore is not valid in the hostname

File: plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMysqlWithoutDataSource.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                .put("connection-password", "password")
+                .build();
+
+        Session session = testSessionBuilder()
+                .setCatalog("mysql")
+                .build();
+
+        DistributedQueryRunner queryRunner = null;
+        try {
+            queryRunner = DistributedQueryRunner.builder(session).build();
```
Comment (findepi) [inline plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMysqlWithoutDataSource.java:47]: Do we really need a DistributedQueryRunner here?
This looks like a unit test of MySqlClient after all

**Review by findepi — APPROVED** [review]
Please make it a unit test

File: plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMysqlWithoutDataSource.java (hunk 2 above)
Comment (kokosing) [reply plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMysqlWithoutDataSource.java:47]: I wanted to see end-to-end how it works. Testing just MySql is not enough as I don't know how given exception will be handled. 

I don't like LocalQueryRunner.

I can use DistributedQueryRunner with single node.

If DistributedQueryRunner is no go, then I don't see a point to add any tests here.

Comment (findepi) [reply plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMysqlWithoutDataSource.java:47]: fine by me without a test --  we have
```
catch (SQLException e) {
            throw new TrinoException(JDBC_ERROR, e);
```

in many places in baseJdbcClient s this is just "the pattern"

**Pull request merged**